remoting/etcdv3/facade.go

```diff
@@ -0,0 +1,81 @@
+package etcdv3
+
+import (
+	"sync"
+	"time"
+)
+
+import (
+	"github.com/apache/dubbo-go/common"
+	"github.com/apache/dubbo-go/common/constant"
+	"github.com/apache/dubbo-go/common/logger"
+)
+
+import (
+	"github.com/dubbogo/getty"
+	perrors "github.com/pkg/errors"
+)
+
+type clientFacade interface {
+	Client() *Client
+	SetClient(*Client)
+	ClientLock() *sync.Mutex
+	WaitGroup() *sync.WaitGroup //for wait group control, zk client listener & zk client container
+	GetDone() chan struct{}     //for zk client control
+	RestartCallBack() bool
+	common.Node
+}
+
+func HandleClientRestart(r clientFacade) {
+
+	var (
+		err       error
+		failTimes int
+	)
+
+	defer r.WaitGroup().Done()
+LOOP:
+	for {
+		select {
+		case <-r.GetDone():
+			logger.Warnf("(ETCDV3ProviderRegistry)reconnectETCDV3 goroutine exit now...")
+			break LOOP
+			// re-register all services
+		case <-r.Client().Done():
+			r.ClientLock().Lock()
```

Comment:
This case will be triggered when client.done is closed.  So where will the client.done be closed?

Reply:
etcdV3Registry.Destroy() -> remoting/etcdv3.Client.Close() ->  remoting/etcdv3.Client.stop()

Reply:
Means only call destory method can trigger reconnect? If the connection of etcd is broken, how can it be reconnected?

Reply:
I have not found the etcdV3Registry.Destroy()function be called except in unit_test.

Reply:
Thx,  i had close remoting/etcdv3/client.Client.ctx. when etcd connection lose.